### PR TITLE
Fix displaying hidden events on threads 

### PR DESCRIPTION
**Pull request by luixxiul**
Fixes https://github.com/vector-im/element-web/issues/22058

This PR updates obsolete styling rules for hidden events in the thread view.

Before:
before

After:
after

On the main timeline: 
after

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix displaying hidden events on threads  ([\#8555](https://github.com/matrix-org/matrix-react-sdk/pull/8555)). Fixes vector-im/element-web#22058. Contributed by @luixxiul.

### CODE REVIEW

**Comment by t3chguy** [comment]
Yes please, it is an oversight that they take more space on the threads timeline

**Comment by t3chguy** [comment]
Also the vertical alignment of hidden & state event avatars regressed, feel free to fix that in the same time

File: res/css/views/right_panel/_ThreadPanel.scss [hunk 1]
```diff
@@ -242,26 +242,22 @@ limitations under the License.
     }
```
Comment (MadLittleMods) [inline res/css/views/right_panel/_ThreadPanel.scss:242]: After this gets some updates from the review suggested, the issue description should get a table of **before** and **after** screenshots to easily compare the changes.

In this case, it's a bit more obvious that it's fixing a broken case [as pictured in the issue](https://github.com/vector-im/element-web/issues/22058) but it sounds like we're also going to fix the main timeline vertical positioning which is way more subtle.

Comment (luixxiul) [reply res/css/views/right_panel/_ThreadPanel.scss:242]: working on that

**Comment by luixxiul** [comment]
The previous commits were removed to restart from scratch.

Since fixing the hidden event view on bubble layout is a bigger task which requires a lot of modification, I would like to leave it for another PR.

**Comment by t3chguy** [comment]
image
image

Any chance you could solve the vertical alignment whilst you're here though? This is Modern layout, both main timeline & thread timeline respectively

File: src/components/views/rooms/EventTile.tsx [hunk 2]
```diff
@@ -1032,6 +1032,13 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
         if (this.context.timelineRenderingType === TimelineRenderingType.Notification) {
             avatarSize = 24;
             needsSenderProfile = true;
+        } else if (isInfoMessage || isInfoMessage &&
+            (this.context.timelineRenderingType === TimelineRenderingType.Thread && !this.props.continuation)
+        ) {
```
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:1037]: This expression might be wrong given that it simplifies to just `isInfoMessage` ignoring the whole second part (`&&` has precedence over `||`)

```suggestion
        } else if (isInfoMessage || isInfoMessage &&
            (this.context.timelineRenderingType === TimelineRenderingType.Thread && !this.props.continuation)
        ) {
```

**Comment by luixxiul** [comment]
> Any chance you could solve the vertical alignment whilst you're here though? This is Modern layout, both main timeline & thread timeline respectively

No problem, I'm going to work on that. I have not touched the rules which govern the alignment on the main timeline yet :smiley:

File: src/components/views/rooms/EventTile.tsx (hunk 2 above)
Comment (luixxiul) [reply src/components/views/rooms/EventTile.tsx:1037]: @t3chguy Would you please add suggestion for me as I am not really sure what the best solution would be? Thanks in advance!

**Comment by t3chguy** [comment]
No but they regressed recently, maybe with your line-height tweaks

File: src/components/views/rooms/EventTile.tsx (hunk 2 above)
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:1037]:
```suggestion
        } else if (isInfoMessage) {
```

This is equivalent to the condition you wrote and I think should be correct

**Comment by t3chguy** [comment]
Hmm in your screenshot the avatar looks to be aligned to the bottom of the event rather than centered

image

**Comment by luixxiul** [comment]
There might be 1px misalignment. Let me check once more... :eyes:

**Comment by luixxiul** [comment]
Not the perfect solution, but it should work..

after

File: res/css/views/rooms/_EventTile.scss [hunk 3]
```diff
@@ -89,20 +89,28 @@ $threadInfoLineHeight: calc(2 * $font-12px); // See: _commons.scss
...
 
-    &.mx_EventTile_info .mx_EventTile_avatar {
-        top: $font-6px;
-        left: $left-gutter;
+    &.mx_EventTile_info {
+        padding-top: 0;
+
+        .mx_EventTile_avatar,
+        .mx_EventTile_e2eIcon {
+            top: 2.25px; // align with mx_EventTile_line
```
Comment (t3chguy) [inline res/css/views/rooms/_EventTile.scss:104]: We really ought to avoid fractional px values, can we not use align-items/align-self/some other vertical alignment tool?

Comment (luixxiul) [reply res/css/views/rooms/_EventTile.scss:104]: Yes definitely. The reason why I chose fractional values is perhaps that all of them required a little bit complicated way. Let me try again.

**Comment by luixxiul** [comment]
Not perfect (this time as well) but this should be better.
 
after

after1

**Review by t3chguy — APPROVED** [review]
Thanks!

**Comment by luixxiul** [comment]
Ideally flexbox should have been applied here to align them, but it required refactoring of the elements around them, so this PR adjusted position with pixel values for now. It should be considered later what the best solution here would be.
